pkg/modulewriter/modulewriter.go

```diff
@@ -355,6 +364,10 @@ func prepDepDir(depDir string, overwrite bool) error {
 		return fmt.Errorf("failed to create directory to save previous deployment groups at %s: %w", prevGroupDir, err)
 	}
 
+	if err := prepArtifactsDir(deploymentio, artifactsDir); err != nil {
```

Comment:
I would put this call before L361 to not split up processing of pervGroupDir. Also now that this is growing all of the prevGroupDir code should also go in its own method `prepPrevGroupDir`.

Reply:
I misread the above block as being the last step of `.ghpc` processing. `prepArtifactsDir` moved up accordingly.